### PR TITLE
Don't re-consume one-time intent extras

### PR DESCRIPTION
**Pull request by amstone326**
Prevent re-consumption of one-time intent extras when an activity is launched from history

### CODE REVIEW

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java [hunk 1]
```diff
@@ -141,10 +142,18 @@
...
 
+    private boolean loginExtraWasConsumed;
+    private static final String EXTRA_CONSUMED_KEY = "login_extra_was_consumed";
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
+        if (savedInstanceState != null) {
+            wasExternal = savedInstanceState.getBoolean(WAS_EXTERNAL_KEY);
```
Comment (phillipm) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: If you set `wasExternal` here then remove it from `processFromExternalLaunch`

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java [hunk 2]
```diff
@@ -274,15 +285,8 @@ public HomeActivityUIController getUiController() {
...
         super.onSaveInstanceState(outState);
-        outState.putBoolean("was_external", wasExternal);
-    }
-
-    @Override
-    protected void onRestoreInstanceState(Bundle inState) {
-        super.onRestoreInstanceState(inState);
-        if (inState.containsKey("was_external")) {
-            wasExternal = inState.getBoolean("was_external");
-        }
```
Comment (phillipm) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: good call

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java (hunk 1 above)
Comment (amstone326) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: Ah, I wasn't paying close enough attention when I refactored this; I think it makes more sense to leave it in processFromExternalLaunch so I'll remove it from here.

**Comment by amstone326** [comment]
@phillipm Addressed comment.

**Pull request merged**